addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -2173,6 +2253,162 @@ def test_compute_allocation_days_after_adding_employee(self):
                 accForm.employee_id = self.employee_hrmanager
 
             updated_allocation = accForm.record
-
             self.assertNotEqual(updated_allocation.number_of_days, allocation_days)
-            self.assertEqual(updated_allocation.number_of_days, 3.0)
+            self.assertAlmostEqual(updated_allocation.number_of_days, 3 * (calendar_emp.work_time_rate / 100), 2)
+
+    @users('test_user')
+    def test_accrual_plan_with_changing_working_schedules_1(self):
+        """
+        Accrual period 1 year, accrual value = 10 days
+        Employee worked from january to june in full time, then from july to december in half time.
+
+        Accrual plan is not based on worked time. Value for the year should be 7.5.
+        0,5 (accrual period) * 10 (accrual value) * 100% = 5 days
+        + 0,5 (accrual period) * 10 (accrual value) * 50% = 2.5 days
+        """
+        if not self._is_hr_contract_installed():
+            return
+
+        accrual_plan = self.env['hr.leave.accrual.plan'].with_context(tracking_disable=True).create({
+            'name': 'Yearly accrual',
+            'carryover_date': 'year_start',
+            'accrued_gain_time': 'end',
+            'is_based_on_worked_time': False,
+            'level_ids': [(0, 0, {
+                'added_value': 10,
+                'added_value_type': 'day',
+                'start_count': 0,
+                'start_type': 'day',
+                'frequency': 'yearly',
+                'cap_accrued_time': False,
+                'action_with_unused_accruals': 'all',
+            })],
```

Comment:
this accrual plan could be defined in the setUp since it's used in multiple places in your tests :)